src/gsy_e/models/strategy/heat_pump.py

```diff
@@ -0,0 +1,171 @@
+from gsy_e.constants import FutureTemplateStrategiesConstants
+from gsy_e.models.state import HeatPumpState
+from gsy_e.models.strategy.energy_parameters.heat_pump import HeatPumpEnergyParameters
+from gsy_e.models.strategy.new_base_strategy import NewStrategyBase
+from gsy_e.models.strategy.order_updater import OrderUpdaterParameters, OrderUpdater
+
+DEFAULT_HEAT_PUMP_ORDER_UPDATE_PARAMS = {
+    AvailableMarketTypes.SPOT: OrderUpdaterParameters(
+        initial_rate=ConstSettings.HeatPumpSettings.BUYING_RATE_RANGE.initial,
+        final_rate=ConstSettings.HeatPumpSettings.BUYING_RATE_RANGE.final,
+        update_interval=duration(
+                minutes=ConstSettings.GeneralSettings.DEFAULT_UPDATE_INTERVAL)),
+    AvailableMarketTypes.FUTURE: OrderUpdaterParameters(
+        initial_rate=FutureTemplateStrategiesConstants.INITIAL_BUYING_RATE,
+        final_rate=FutureTemplateStrategiesConstants.FINAL_BUYING_RATE,
+        update_interval=duration(
+            minutes=FutureTemplateStrategiesConstants.UPDATE_INTERVAL_MIN)),
+}
+
+if TYPE_CHECKING:
+    from gsy_e.models.market import MarketBase
+
+
+class HeatPumpStrategy(NewStrategyBase):
+    """Strategy for heat pumps with storages."""
+    # pylint: disable=too-many-arguments)
+    def __init__(self,
+                 maximum_power_rating: float = ConstSettings.HeatPumpSettings.MAX_POWER_RATING_KW,
+                 min_temperatur_C: float = ConstSettings.HeatPumpSettings.MIN_TEMP_C,
+                 max_temperatur_C: float = ConstSettings.HeatPumpSettings.MAX_TEMP_C,
+                 external_temperatur_C: float = ConstSettings.HeatPumpSettings.EXT_TEMP_C,
+                 tank_volume_l: float = ConstSettings.HeatPumpSettings.TANK_VOL_L,
+                 average_consumption: float =
+                 ConstSettings.HeatPumpSettings.AVERAGE_CONSUMPTION_KW,
+                 source_type: int = ConstSettings.HeatPumpSettings.TYPE,
+                 order_updater_parameters: Dict[
+                     AvailableMarketTypes, OrderUpdaterParameters] = None,
+                 preferred_buying_rate: float =
+                 ConstSettings.HeatPumpSettings.PREFERRED_BUYING_RATE
+                 ):
+        if not order_updater_parameters:
+            order_updater_parameters = DEFAULT_HEAT_PUMP_ORDER_UPDATE_PARAMS
+
+        super().__init__(order_updater_parameters=order_updater_parameters)
+
+        self._energy_params = HeatPumpEnergyParameters(
+            maximum_power_rating=maximum_power_rating,
+            min_temperatur_C=min_temperatur_C,
+            max_temperatur_C=max_temperatur_C,
+            external_temperatur_C=external_temperatur_C,
+            tank_volume_l=tank_volume_l,
+            average_consumption=average_consumption,
+            source_type=source_type
+        )
+
+        self.preferred_buying_rate = preferred_buying_rate
+        self._markets_buffer = []  # TODO: discuss if this buffer is really necessary
```

Comment:
I discussed with @spyrostz yesterday, the HeatPump will not be taking part in the future markets for now, because the accounting of the storage is complicated for future time slot. It also does not seem that the design includes trading in the future markets. So, we decided, that the HeatPump will only trade in the spot market, which will simplify this strategy implementation a lot. With this we do not need the `self._markets_buffer`.

Reply:
I already did this here: [6e248d9](https://github.com/gridsingularity/gsy-e/pull/1568/commits/6e248d9ce74f332c47d5f35108e15b1639e2ecbd)